addons/account/models/account_move.py

```diff
@@ -2266,7 +2266,8 @@ def is_internal_partner(partner):
             if is_internal_partner(partners[0]):
                 # Search for partners in the mail's body.
                 body_mail_addresses = set(email_re.findall(msg_dict.get('body')))
-                partners = [partner for partner in self._mail_find_partner_from_emails(body_mail_addresses, extra_domain) if not is_internal_partner(partner)]
+                company_id = custom_values.get('company_id', self.env.company.id)
+                partners = [partner for partner in self._mail_find_partner_from_emails(body_mail_addresses, extra_domain) if not is_internal_partner(partner) and partner.company_id.id in (False, company_id)]
```

Comment:
```suggestion
                partners = [
                    partner 
                    for partner in self._mail_find_partner_from_emails(body_mail_addresses, extra_domain) 
                    if not is_internal_partner(partner) and partner.company_id.id in (False, company_id)]
                ]

Reply:
done, thanks
(sorry I didn't see the message earlier)